freeipa/src/test/java/com/sequenceiq/freeipa/flow/stack/provision/handler/ClusterProxyRegistrationHandlerTest.java

```diff
@@ -0,0 +1,104 @@
+package com.sequenceiq.freeipa.flow.stack.provision.handler;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import java.util.Set;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import com.sequenceiq.flow.event.EventSelectorUtil;
+import com.sequenceiq.freeipa.entity.InstanceMetaData;
+import com.sequenceiq.freeipa.flow.stack.provision.event.clusterproxy.ClusterProxyRegistrationFailed;
+import com.sequenceiq.freeipa.flow.stack.provision.event.clusterproxy.ClusterProxyRegistrationRequest;
+import com.sequenceiq.freeipa.flow.stack.provision.event.clusterproxy.ClusterProxyRegistrationSuccess;
+import com.sequenceiq.freeipa.service.stack.ClusterProxyService;
+import com.sequenceiq.freeipa.service.stack.instance.InstanceMetaDataService;
+
+import reactor.bus.Event;
+import reactor.bus.EventBus;
+
+@ExtendWith(MockitoExtension.class)
+class ClusterProxyRegistrationHandlerTest {
+
+    @Mock
+    private EventBus eventBus;
+
+    @Mock
+    private ClusterProxyService clusterProxyService;
+
+    @Mock
+    private InstanceMetaDataService instanceMetaDataService;
+
+    @InjectMocks
+    private ClusterProxyRegistrationHandler underTest;
+
+    @Test
+    public void testSelector() {
+        assertEquals(EventSelectorUtil.selector(ClusterProxyRegistrationRequest.class), underTest.selector());
+    }
+
```

Comment:
I can, but CP registration without instances doesn't make too much sense. It would fail during the registration, doesn't matter which one is called. So it has no real value at this point, but if you still think it worth it, I will add the test

Reply:
I see. Maybe would it make sense to add a check in the handler if there are instancemetadata at all, and if not then registration should not be attempted? I do not know under what circumstances that would happen though.

Reply:
ok, I changed the logic. this should be fine now